plugins/woocommerce-admin/client/marketing/data-multichannel/resolvers.ts

```diff
@@ -114,3 +116,19 @@ export function* getCampaigns( page: number, perPage: number ) {
 		throw error;
 	}
 }
+
+export function* getCampaignTypes() {
+	try {
+		const data: CampaignType[] = yield apiFetch( {
+			path: `${ API_NAMESPACE }/campaign-types`,
+		} );
+
+		yield receiveCampaignTypes( data );
+	} catch ( error ) {
+		if ( isApiFetchError( error ) ) {
+			yield receiveCampaignTypes( error );
+		}
+
+		throw error;
```

Comment:
> Could it be due to the underlying WP environment? I'm using WP 6.1.1.

I was using 5.9.5.

Just found the `@wordpress/data` resolvers started catching errors since [version 6.4.0](https://github.com/WordPress/gutenberg/blob/%40wordpress/data%406.4.0/packages/data/src/redux-store/index.js#L431-L452), and it's included in WP >= 6.0. See:
- https://github.com/WordPress/wordpress-develop/blob/5.9.5/package.json#L94
- https://github.com/WordPress/wordpress-develop/blob/6.0.0/package.json#L93

Considering [the latest WC still states it supports WP 5.9](https://github.com/woocommerce/woocommerce/blob/7.5.0-rc.1/plugins/woocommerce/woocommerce.php#L11) and this feature won't use `hasResolutionFailed` or `getResolutionError` to get the thrown errors, probably it would be better not to re-throw errors from the resolvers.

Reply:
> Considering [the latest WC still states it supports WP 5.9](https://github.com/woocommerce/woocommerce/blob/7.5.0-rc.1/plugins/woocommerce/woocommerce.php?rgh-link-date=2023-03-10T07%3A59%3A56Z#L11) and this feature won't use `hasResolutionFailed` or `getResolutionError` to get the thrown errors, probably it would be better not to re-throw errors from the resolvers.

Yeah, that makes sense, to make it work with WP 5.9.

For reference, WP 5.9 uses `@wordpress/data@6.1.5`, and here's the corresponding code, there is no try/catch block: https://github.com/WordPress/gutenberg/blob/1081a28b2368fcebd9a14554fbd0e25eb33fbcb6/packages/data/src/redux-store/index.js#L414-L428

TIL about `hasResolutionFailed` or `getResolutionError`. Thanks for sharing that! Seems like it is introduced in PR https://github.com/WordPress/gutenberg/pull/38669. There should be more documentation around that. In the future we don't need to manually store the error on our own like what I did in this project. 😂 

I have removed the `throw error;` line for `getCampaignTypes` resolver in [`1fea018` (#37044)](https://github.com/woocommerce/woocommerce/pull/37044/commits/1fea0185fe0002524bd7cf2a28dc915e8f9d0aa2). For other resolvers, I shall do it in separate PR.